### PR TITLE
DOC: reshuffle stats distributions documentation 

### PR DESCRIPTION
**Pull request by ev-br**
Try making the docs of univariate distributions a bit clearer, esp for people without much experience with scipy.stats.
- tweak the docstrings of rv_continuous and rv_discrete, add some basic information
- move the lists of methods of distributions out of the module docstring
- tweak the generated docstrings of individual distributions, remove mysterious statements and details.
  Specifically, I'm removing the Parameters section from the docstrings of distributions. IMO, this section as it was was more confusing than helpful (are these parameters to what?). Also, when this section gone, a section with a PDF is better visible.

### CODE REVIEW

File: scipy/stats/_distn_infrastructure.py [hunk 1]
```diff
@@ -1279,108 +1287,69 @@ class rv_continuous(rv_generic):
...
+    expect
+    median
+    mean
+    std
+    var
+    interval
+    __call__
+    fit
+    fit_loc_scale
+    nnlf
```
Comment (ev-br) [inline scipy/stats/_distn_infrastructure.py]: Notice that I'm adding three fit-related methods here, which were not listed as public methods before.

File: scipy/stats/_distn_infrastructure.py [hunk 2]
```diff
@@ -1409,55 +1402,44 @@ class rv_continuous(rv_generic):
...
-                ...
-            ...
+    >>> class gaussian_gen(rv_continuous):
+    ...     "Gaussian distribution"
+    ...     def _pdf(self, x):
+    ...         return np.exp(-x**2 / 2.) / np.sqrt(2.0 * np.pi)
+    >>> gaussian = gaussian_gen(name='gaussian')
 
-    """
+    ``scipy.stats`` distributions are *instances*, so we here we subclass
```
Comment (rgommers) [inline scipy/stats/_distn_infrastructure.py]: typo "we here we"

File: scipy/stats/_distn_infrastructure.py [hunk 3]
```diff
@@ -1279,108 +1301,69 @@ class rv_continuous(rv_generic):
...
+    Whether a shape parameter is valid is decided by an ``_argcheck`` method
+    (which defaults to checking that its arguments are strictly positive.)
 
-    ``median(<shape(s)>, loc=0, scale=1)``
-        Median of the distribution.
+    **Subclassing**
 
-    ``mean(<shape(s)>, loc=0, scale=1)``
-        Mean of the distribution.
+    New random variables can be defined by subclassing `rv_continuous` class
```
Comment (rgommers) [inline scipy/stats/_distn_infrastructure.py]: textual: either ``the `rv_continuous` class`` or `` `rv_continuous` ``.

File: scipy/stats/_distn_infrastructure.py (hunk 1 above)
Comment (rgommers) [inline scipy/stats/_distn_infrastructure.py]: The ``fit`` docstring has an issue with the list of values that can be used for ``kwds``; html is messed up (optimizer is bold). Might want to take that along here.

**Comment by rgommers** [comment]
This looks great! I checked the rendered html and it's indeed much cleaner and more user-friendly now.

**Comment by ev-br** [comment]
Addressed the review comments.

And some github magic: closes https://github.com/scipy/scipy/issues/4526

**Comment by ev-br** [comment]
ping @romunov @aconley  --- you guys have recently looked at this; feedback on how to improve this further would be appreciated.

**Comment by rgommers** [comment]
Merged, thanks Evgeni!

**Comment by rgommers** [comment]
Suggestions for further improvements still welcome of course!

**Comment by aconley** [comment]
I guess I'm a little late to the discussion -- but I'm definitely appreciative of any documentation improvements!

**Pull request merged**